File: Programmers/2/70129. 이진 변환 반복하기/이진 변환 반복하기.java
import java.util.*;

class Solution {
    public int[] solution(String s) {
        int transformCount = 0;
        int zeroCount = 0;

        while (!s.equals("1")) {
            int zeros = 0;
            
            for (char c : s.toCharArray()) {
                if (c == '0') zeros++;
            }
            
            zeroCount += zeros;
            
            int ones = s.length() - zeros;
            
            s = Integer.toBinaryString(ones);
            
            transformCount++;
        }

        return new int[]{transformCount, zeroCount};
    }
}

File: Programmers/2/70129. 이진 변환 반복하기/이진 변환 반복하기.cpp
#include <string>
#include <vector>

using namespace std;

vector<int> solution(string s) {
    int transformCount = 0;
    int removedZeroCount = 0;

    while (s != "1")
    {
        int zeros = 0;
        
        for (char c : s)
        {
            if (c == '0') zeros++;
        }
        removedZeroCount += zeros;

        int ones = s.size() - zeros;
        
        string binary = "";
        
        while (ones > 0)
        {
            binary = char('0' + (ones % 2)) + binary;
            ones /= 2;
        }

        s = binary;
        transformCount++;
    }

    return {transformCount, removedZeroCount};
}